tests/lib/io/test_keys.py

```diff
@@ -95,7 +95,7 @@ class MyDataRoot(DataRoot):
     expected = MetadataType('metadata', mdr._root, mdr._metadata_format)
     assert mdr.metadata == expected
 
-    expected_key = DatasetKey(mdr._root, mdr._data_format, 'saturn', 'draw_0', 'cassini')
+    expected_key = DatasetKey(mdr._root, mdr._data_format, 'saturn', 'draw_0', 'cassini', None)
```

Comment:
I assume `None` is the default which returns all columns.

IMO you should omit this explicit `None` and have `DatasetKey` supply a default value.

```python
>>> class Keys(typing.NamedTuple):
...   name: str
...   columns: typing.Optional[str] = None                                                                                                                                                                                                     
...                                                                                                                                                                                                                                            
>>> k1 = Keys(name="data")
>>> k2 = Keys(name="other data", columns=['foo', 'bar'])                                                                                                                                                                                       
>>> k1
Keys(name='data', columns=None)                                                                                                                                                                                                                
>>> k2                                                                                                                                                                                                                                         
Keys(name='other data', columns=['foo', 'bar'])
```